pkg/lib/infra/redisqueue/producer.go

```diff
@@ -0,0 +1,104 @@
+package redisqueue
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	goredis "github.com/go-redis/redis/v8"
+
+	"github.com/authgear/authgear-server/pkg/lib/infra/redis/appredis"
+	"github.com/authgear/authgear-server/pkg/util/base32"
+	"github.com/authgear/authgear-server/pkg/util/clock"
+	corerand "github.com/authgear/authgear-server/pkg/util/rand"
+)
+
+const (
+	idAlphabet string = base32.Alphabet
+	idLength   int    = 32
+)
+
+var rng *rand.Rand = corerand.SecureRand
+
+var TTL = 24 * time.Hour
```

Comment:
It seems there shouldn't be TTL on the tasks. They should be processed if they exist.

Reply:
We need TTL so that it will be deleted automatically. We do not store them indefinitely.

Reply:
🤔 Finished tasks can be removed automatically, However, it seems incompleted tasks shouldn't be deleted.

Or actually the task result should be stored into db?
And we can also keep track the history of import tasks. One days seems too short for a import result?
And as it is read only it shouldn't cause large loadings on the db?

Reply:
> Finished tasks can be removed automatically, However, it seems incompleted tasks shouldn't be deleted.

If the task is not dequeued within 1 day, then yes, it is a problem. Otherwise, the task will be saved with status update with a TTL of 24 hours. For our usage on hand, it should be fine.

> actually the task result should be stored into db?
I tend to avoid housekeeping the task results.

> One days seems too short for a import result?
Auth0 also keeps the result for 1 day https://auth0.com/docs/manage-users/user-migration/bulk-user-imports#job-timeouts

> Or maybe we need audit logs?
We can add it later.

Reply:
My concern is if there is any problem occurred causing the consumer stopped to consume tasks from the queue, and if the tasks will disappear automatically after one day, it could be difficult for us to discover the problem.

Reply:
The consumer now does not recover from panic so a panic will cause it to stop. Let me fix it.

Reply:
I think there could be other unknown reason / bugs that causing the consumer to stop consuming tasks, so I guess we might want some kind of monitoring to ensure the consumer is working. I think we can open another issue for it?

Reply:
https://github.com/authgear/authgear-server/issues/4025